torch/csrc/autograd/variable.cpp

```diff
@@ -21,91 +21,18 @@
 
 namespace torch {
 namespace autograd {
-Variable::Impl::Impl(at::Tensor data, std::unique_ptr<Variable::AutogradMeta> autograd_meta, bool requires_grad, Edge gradient_edge)
-    : TensorImpl(data.type_id(), data.dtype(), /*allocator=*/nullptr, /* is variable */ true),
-      data_(std::move(data)) {
-  autograd_meta->grad_fn_ = std::move(gradient_edge.function);
-  autograd_meta->requires_grad_ = false;
-  autograd_meta->is_view_ = false;
-  autograd_meta->output_nr_ = gradient_edge.input_nr;
-  autograd_meta->pyobj_ = nullptr;
+Variable::AutogradMeta::AutogradMeta(at::TensorImpl* self_impl, bool requires_grad, Edge gradient_edge) {
```

Comment:
This lost the assert that `self_impl` is defined?

Reply:
Added below.

Reply:
@gchanan I checked and confirmed that all call sites of the `AutogradMeta` constructor have checks (e.g. https://github.com/pytorch/pytorch/pull/17072/files#diff-e8b542b35947b93dc1b8ed6babcc4dc7R538) for whether the `self_impl` parameter is defined, so we don't need to perform the check here.